src/types.rs

```diff
@@ -1,5 +1,65 @@
 use std::collections::HashMap;
 
+use std::fmt;
+use std::cmp;
+use serde::*;
+use serde::de::{Visitor, Error, Unexpected};
+
+#[derive(Clone, Debug)]
+pub struct Timestamp {
+    string_repr: String,
+    float_repr: f64,
+}
+
```

Comment:
Since the struct methods are private, we don't have any way for the consumers to actually inspect the timestamp.